src/parser.rs

```diff
@@ -1663,6 +1664,29 @@ impl Parser {
         })
     }
 
+    pub fn parse_show(&mut self) -> Result<Statement, ParserError> {
+        if self.parse_keyword("COLUMNS") {
+            self.parse_show_columns()
+        } else {
+            self.expected("COLUMNS", self.peek_token())
+        }
+    }
+
+    fn parse_show_columns(&mut self) -> Result<Statement, ParserError> {
+        if !self.parse_keyword("FROM") {
+            return self.expected("FROM", self.peek_token());
+        }
+        let table_or_schema = self.parse_object_name()?;
+
+        if let Some(tok) = self.peek_token() {
+            // TODO: show columns from .. from .. is unsupported
+            return self.expected("Nothing", tok.into());
+        }
```

Comment:
tiny optional nit: I think the convention elsewhere is to not include special handling for extraneous stuff and let the syntax error occur on the semicolon.

Reply:
Oh does the parser auto-error if it doesn't consume everything? Okay.

Reply:
Yep, exactly.